lib/Support/Errors.cpp

```diff
@@ -0,0 +1,46 @@
+/// \file Errors.cpp
+/// \brief this file contains the various errors thrown by revng
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Support/Errors.h"
+
+char revng::DiffError::ID;
+
+std::error_code revng::DiffError::convertToErrorCode() const {
+  return llvm::inconvertibleErrorCode();
+}
+
+void revng::DiffError::log(llvm::raw_ostream &OS) const {
+  OS << "Diff Errors\n";
+  for (auto &Reason : Reasons) {
+    OS << "\t" << Reason << "\n";
+  }
+}
+
+llvm::Error revng::DiffError::concatenate(llvm::Error SourceError,
+                                          llvm::StringRef NewReason) {
+  if (not SourceError.isA<DiffError>())
+    return SourceError;
+
+  llvm::Error ToReturn = llvm::Error::success();
+  return llvm::handleErrors(std::move(SourceError),
+                            [&](std::unique_ptr<revng::DiffError> DiffError) {
+                              DiffError->Reasons.push_back(NewReason.str());
+                              return llvm::Error(std::move(DiffError));
+                            });
+}
+
+void revng::DiffError::addReason(llvm::Error Error) {
+  if (!Error)
+    return;
+
+  std::string S;
+  llvm::raw_string_ostream OS(S);
+  OS << Error;
+  OS.flush();
+  llvm::consumeError(std::move(Error));
+  addReason(S);
+}
```

Comment:
Here we're slurping *any* errors AFAIU. I think we should slurp only errors of the same type.
Otherwise this is not `DiffError` but `ErrorList` again.

Reply:
Uhm, seems like your design is to reuse this for non-diff-related purposes.
Do you want to use it for, say, reporting validation errors of `Binary::verify`? If so, I'm fine with keeping this in `librevngSupport` but it needs to be renamed to... `StringErrorList`?

Reply:
probably jack has some opinions too about this